server/routes/journeys/develop-an-initial-plan/check-answers/controller.ts

```diff
@@ -0,0 +1,61 @@
+import { NextFunction, Request, Response } from 'express'
+import { BaseJourneyController } from '../../base/controller'
+import { SanitisedError } from '../../../../sanitisedError'
+
+export class PlanCheckAnswersController extends BaseJourneyController {
+  GET = async (req: Request, res: Response) => {
+    req.journeyData.isCheckAnswers = true
+
+    const { plan } = req.journeyData
+    const sortedNeeds = plan!.identifiedNeeds?.sort((needA, needB) => {
+      return new Date(needA.createdDate).getTime() - new Date(needB.createdDate).getTime()
+    })
+    res.render('develop-an-initial-plan/check-answers/view', {
+      plan: {
+        ...plan,
+        identifiedNeeds: sortedNeeds,
+      },
+      csipRecordUrl: `/csip-records/${req.journeyData.csipRecord!.recordUuid}`,
+    })
+  }
+
+  checkSubmitToAPI = async (req: Request, res: Response, next: NextFunction) => {
+    const plan = req.journeyData.plan!
+    try {
+      await this.csipApiService.createPlan(req, {
+        caseManager: plan.caseManager!,
+        firstCaseReviewDate: plan.firstCaseReviewDate!,
+        reasonForPlan: plan.reasonForPlan!,
+        identifiedNeeds: plan.identifiedNeeds!.map(
+          ({ closedDate, createdDate, identifiedNeed, intervention, progression, responsiblePerson, targetDate }) => ({
```

Comment:
Not a fan of destructuring when there are so many properties inside it. Makes the line long and unruly. I have the following in my branch, also addresses bill's comments below

```
      await this.csipApiService.createInitialPlan(req, {
        caseManager: plan.caseManager!,
        firstCaseReviewDate: plan.firstCaseReviewDate!,
        reasonForPlan: plan.reasonForPlan!,
        identifiedNeeds: plan.identifiedNeeds!.map(identifiedNeed => ({
          identifiedNeed: identifiedNeed.identifiedNeed!,
          responsiblePerson: identifiedNeed.responsiblePerson!,
          intervention: identifiedNeed.intervention!,
          createdDate: todayString(),
          targetDate: identifiedNeed.targetDate!,
          ...(identifiedNeed.progression ? { progression: identifiedNeed.progression! } : {}),
        })),
      })
   ```

Reply:
or maybe 
```
        identifiedNeeds: plan.identifiedNeeds!.map(identifiedNeed => ({
          ...identifiedNeed,
          createdDate: todayString(),
          ...(identifiedNeed.progression ? { progression: identifiedNeed.progression! } : {}),
        })),
```

the type will be wrong tho lol

Reply:
Hi @JackReeveMoJ I've removed closedDate from this list of properties as we didn't want it, so the total line length is now shorter - I think it would end up being more verbose and less easy to read if I didn't destructure with what's there now. Is that ok, or do you still not like the destructuring?

Reply:
I still don't like the destructuring tbh but I'm not gonna block the PR for it if you feel its easier to read this way